test/jit/test_backends.py

```diff
@@ -69,47 +91,123 @@ def compare_py_jit_backend(self, name, input):
         self.assertEqual(python_output, backend_output)
         self.assertEqual(jit_output, backend_output)
 
-    @skipIfRocm
-    def test_simple(self):
+    def save_load(self):
         """
-        This is a simple test that compiles MyModule for the test backend and ensures it produces the correct
-        answers for each method.
+        Save and load the lowered module.
         """
+        # Save the lowered module.
+        buffer = io.BytesIO()
+        torch.jit.save(self.lowered_module, buffer)
+
+        # Load the lowered module.
+        buffer.seek(0)
+        self.lowered_module = torch.jit.load(buffer)
+
+
+class BasicModuleTest(JitBackendTestCase):
+    """
+    Tests for MyModule.
+    """
+
+    def setUp(self):
+        super().setUp()
+        # Create Python, JIT and backend versions of MyModule.
+        self.module = MyModule()
+        self.scripted_module = torch.jit.script(MyModule())
+        self.lowered_module = to_test_backend_multi(
+            self.scripted_module._c,
+            {"accum": {"": ""}, "sub_accum": {"": ""}, "forward": {"": ""}},
+        )
+
+    def test_execution(self):
         # Test execution with backend against Python and JIT.
         input = torch.randn(5)
 
         # Test all three module methods.
-        self.compare_py_jit_backend("accum", input)
-        self.compare_py_jit_backend("sub_accum", input)
-        self.compare_py_jit_backend("forward", input)
+        self.check_function("accum", input)
+        self.check_function("sub_accum", input)
+        self.check_function("forward", input)
 
     @skipIfRocm
     def test_save_load(self):
-        """
-        This method tests that a lowered module till produces the same output as a Python module and ScriptModule after
-        saving and loading.
-        """
-        # Save the lowered module.
-        buffer = io.BytesIO()
-        torch.jit.save(self.lowered_module, buffer)
+        # Lowered module should produce the same outputs.
+        self.test_execution()
 
         # Save the compile spec to compare against the version retrieved after loading.
         pre_compile_spec = self.lowered_module.__getattr__("__method_compile_spec")
 
-        # Load the lowered module.
-        buffer.seek(0)
-        self.lowered_module = torch.jit.load(buffer)
+        # Save and load the lowered module.
+        self.save_load()
 
         # Get the compile spec after loading.
         post_compile_spec = self.lowered_module.__getattr__("__method_compile_spec")
 
         # Compile specs should match.
         self.assertEqual(pre_compile_spec, post_compile_spec)
 
+        # Loaded module should produce the same outputs.
+        self.test_execution()
+
+
+class NestedModuleTest(JitBackendTestCase):
```

Comment:
It's kind of hard to read this test because `NestedModule` and `MyModule` are defined at the top of this file. Generally we define any necessary test-specific fixtures inside the body of a test so that things stay local.

Reply:
It's hard to nest these because this test and the other test both use `MyModule`.